### DISCUSSION TITLE
Styling keyboard focus differently than mouse focus with useSelect?

### DISCUSSION THREAD

**Discussion by seanrcollings** [adobe/react-spectrum#6858] [Q&A] [answered]
I'm trying to figure out how to style a select listbox differently when it's being navigated with keyboard vs when it's being navigated with the mouse.

[CodeSandbox Link](https://codesandbox.io/p/sandbox/heuristic-frog-7ymwng?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522clznje6tl00072e69lk4lyffh%2522%252C%2522sizes%2522%253A%255B100%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522clznje6tl00032e692e9ut6n7%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522clznje6tl00042e697kxdpohq%2522%257D%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522clznje6tl00062e697be06ntr%2522%257D%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522clznje6tl00032e692e9ut6n7%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clznje6tl00022e69j6eomcpk%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252FComboBox.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522clznjf9sr001f2e69bhz6ha8f%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A90%252C%2522startColumn%2522%253A3%252C%2522endLineNumber%2522%253A90%252C%2522endColumn%2522%253A3%257D%255D%252C%2522filepath%2522%253A%2522%252Fsrc%252FListBox.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%255D%252C%2522id%2522%253A%2522clznje6tl00032e692e9ut6n7%2522%252C%2522activeTabId%2522%253A%2522clznjf9sr001f2e69bhz6ha8f%2522%257D%252C%2522clznje6tl00062e697be06ntr%2522%253A%257B%2522id%2522%253A%2522clznje6tl00062e697be06ntr%2522%252C%2522activeTabId%2522%253A%2522clznjg9mu00132e685hx6r4oz%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522clznje6tl00052e692jgjvvut%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%257D%252C%257B%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522id%2522%253A%2522clznjg9mu00132e685hx6r4oz%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%257D%252C%2522clznje6tl00042e697kxdpohq%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522clznje6tl00042e697kxdpohq%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Atrue%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

In the above example I'm using the `useFocusRing` hook. I noticed that if you begin navigating with the keyboard, `isFocusVisible` is correctly set to true & the border styles are applied. However if you then start navigating with the mouse, `isFocusVisible` remains true and the border doesn't go away.

Does anyone know if there's an alternative solution that resolves this issue? 

Thanks!

**Comment by snowystinger** [comment, marked as answer]
We don't know if you've bumped the mouse accidentally or are actually switching to interact with the mouse as your primary interaction method. So we opt to leave it in keyboard modality until a users actually clicks something.
